vermouth/rcsu/get_go.py

```diff
@@ -0,0 +1,164 @@
+            return True
+    return False
+
+class GetGo():
+    """
+    Generate the Go model interaction parameters.
+    """
+    def __init__(self,
+                 contact_map,
+                 cutoff_short,
+                 cutoff_long,
+                 go_eps,
+                 res_dist,
+                 res_graph,
+                 regions=None,
+                 water_bias=None,
+                 idp_water_bias=None,
+                 prefix="VS"):
+
+        self.contact_map = contact_map
+        self.cutoff_short = cutoff_short
+        self.cutoff_long = cutoff_long
+        self.go_eps = go_eps
+        self.res_dist = res_dist
+        self.itp_file = open('GoVirtIncludes.itp', "w", encoding='UTF-8')
+        self.prefix = prefix
+        self.res_graph = res_graph
+        self.water_bias = water_bias
+        self.idp_water_bias = idp_water_bias
+        self.regions = regions
+
+    def write_go_structural_bias(self, molecule):
+        chain_id_to_resnode = {}
+        # find all pairs of residues that are within bonded distance of
+        # self.res_dist
+        connected_pairs = dict(nx.all_pairs_shortest_path_length(self.res_graph,
+                                                            cutoff=self.res_dist))
+
+        # for each residue collect the chain and residue in a dict
+        # we use this later for identifying the residues from the
+        # contact map
+        for resnode in self.res_graph.nodes:
+            chain = self.res_graph.nodes[resnode].get('chain', None)
+            # in vermouth within a molecule all resid are unique
+            # when merging multiple chains we store the old resid
+            # the go model always references the input resid i.e.
+            # the _old_resid
+            resid = self.res_graph.nodes[resnode].get('_old_resid')
+            chain_id_to_resnode[(chain, resid)] = resnode
+
+        for contact in self.contact_map:
+            resIDA, chainA, resIDB, chainB = contact
+            # identify the contact in the residue graph based on
+            # chain ID and resid
+            resA = chain_id_to_resnode[(chainA, resIDA)]
+            resB = chain_id_to_resnode[(chainB, resIDB)]
+            # make sure that both residues are not connected
+            # note: contacts should be symmteric so we should be
+            # able to get rid of the second if clause
+            if resB not in connected_pairs[resA] and resA not in connected_pairs[resB]:
+                # now we lookup the backbone nodes within the residue contact
+                bb_node_A = _get_bb_nodes(molecule, self.res_graph.nodes[resA]['graph'].nodes)
+                bb_node_B = _get_bb_nodes(molecule, self.res_graph.nodes[resB]['graph'].nodes)
+                # compute the distance between bb-beads
+                dist = np.linalg.norm(molecule.nodes[bb_node_A]['position'] -
+                                      molecule.nodes[bb_node_B]['position'])
```

Comment:
I have the feeling we should make our own cartesian/spatial module with these kind of staple functions. That'll make plugging in PBC at a later date easier.
Future TODO though

Reply:
let's keep this in mind as a todo indeed